packages/cadl-ranch-specs/http/encode/bytes/main.tsp

```diff
@@ -0,0 +1,109 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Test for encode decorator on bytes.")
+@supportedBy("dpg")
+@scenarioService("/encode/bytes")
+namespace Encode.Bytes;
+
+@encode(BytesKnownEncoding.base64)
+scalar base64Bytes extends bytes;
+
+@route("/query")
+@scenario
+@scenarioDoc("""
+Test different encode for bytes query parameters.
+Expected query parameter: 
+default=dGVzdA== (base64 encode of test)
+base64=dGVzdA==
+base64url=dGVzdA (base64url encode of test)
+base64-array=dGVzdA==, dGVzdA==
+""")
+op query(
```

Comment:
Can we have one scenario per encoding style like it is in the encode/duration spec

Reply:
it would be nice that the dashboard shows that you support each encoding independently. 

Seeing just that it is supported in the query but not seeing we can encode bytes as base64, base64url feels limited

Reply:
I see. Is there any better way to define these operations? Decorator could not be used in template param.

Reply:
hhm, what were you trying to reuse, it feels like we do have unfortunatelly  do a lot of duplication

Reply:
Separated. You may see there are many duplication; I did not find good way to reduce them.